File: Practice/src/coding_basic_training/ConcatenatedNumbers.java
package coding_basic_training;

public class ConcatenatedNumbers {
//	이어 붙인 수
	public int solution(int[] num_list) {
		String a = "", b = "";
		for (int i = 0; i < num_list.length; i++) {
			if (num_list[i] % 2 == 0) {
				a += num_list[i];
			} else {
				b += num_list[i];
			}
		}
		return Integer.parseInt(a) + Integer.parseInt(b);
	}
}

File: Practice/src/coding_basic_training/ConcatenatedNumbers.js
function solution(num_list) {
    let a = "";
    let b = "";
    for (let i of num_list) {
        if (i % 2 == 0) {
            a += i;
        } else {
            b += i;
        }
    }
    return Number(a) + Number(b);
    // reduce를 이용한 방법
    // const { odds, evens } = num_list.reduce(
    //     ({ odds, evens }, num) => {
    //         if (num % 2 === 0) {
    //             evens.push(num);
    //         } else {
    //             odds.push(num);
    //         }
    //         return { odds, evens };
    //     },
    //     { odds: [], evens: [] }
    // );
    // return Number(odds.join("")) + Number(evens.join(""));
}
